### PR TITLE
Improve OnBuild detection by looking into Image metadata

### PR DESCRIPTION
**Pull request by mfojtik**
This makes the detection based on the Image tag name obsolete.

### CODE REVIEW

**Comment by mfojtik** [comment]
[test]

**Comment by mfojtik** [comment]
@soltysh @bparees PTAL

**Comment by mfojtik** [comment]
Upstream PR: https://github.com/fsouza/go-dockerclient/pull/212

File: pkg/build/strategies/strategies.go [hunk 1]
```diff
@@ -1,23 +1,51 @@
...
+		return &result, nil
+	}
+	image, err := d.PullImage(request.BaseImage)
+	if err != nil {
+		return nil, err
+	}
+	result.Image = image
+	result.OnBuild = d.IsImageOnBuild(request.BaseImage)
+	return &result, nil
+}
```
Comment (soltysh) [inline pkg/build/strategies/strategies.go]: This look like big nasty copy&paste :stuck_out_tongue_winking_eye: , how about replacing the code starting from line 35-50 with something like:
```
var image *docker.Image
if request.ForcePull {
    image, err := d.PullImage(request.BaseImage)
} else {
    image, err := d.CheckAndPull(request.BaseImage)
}
if err != nil {
    return nil, err
}
result.Image = image
result.OnBuild = d.IsImageOnBuild(request.BaseImage)
return &result, nil
```

Comment (mfojtik) [inline pkg/build/strategies/strategies.go]: yeah, I'm gettin lazy these days ;-) I must done some refactor in docker.go and I forgot to make this func nice ;-)

**Comment by mfojtik** [comment]
@soltysh fixed :-)

**Comment by soltysh** [comment]
Perfect, LGTM.

**Comment by mfojtik** [comment]
[merge]

**Pull request merged**